### PR TITLE
Adds bindings for the world convenience class

### PR DESCRIPTION
**Pull request by Voldivh**
# 🎉 New feature

## Summary
This PR creates the python bindings for the `World` convenience class.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/gz/sim/World.cc [hunk 1]
```diff
@@ -22,23 +22,79 @@
...
-    "gravity", &gz::sim::World::Gravity,
-    "Get the gravity in m/s^2.");
+  py::class_<gz::sim::World>(module, "World")
+  .def(py::init<gz::sim::Entity>())
+  .def("entity", &gz::sim::World::Entity,
+      "Get the entity which this World is related to.")
+  .def("valid", &gz::sim::World::Valid,
+      py::arg("ecm"),
+      "Check whether this world correctly refers to an entity that"
+      "has a components::World.")
```
Comment (ahcorde) [inline python/src/gz/sim/World.cc:42]:
```suggestion
      "Check whether this world correctly refers to an entity that "
      "has a components::World.")
```

File: python/src/gz/sim/World.cc [hunk 2]
```diff
@@ -22,23 +22,79 @@
...
+      "Get spherical coordinates for the world origin.")
+  .def("set_spherical_coordinates", &gz::sim::World::SetSphericalCoordinates,
+      py::arg("ecm"),
+      py::arg("spherical_coordinates"),
+      "Set spherical coordinates for the world origin.")
+  .def("light_by_name", &gz::sim::World::LightByName,
+      py::arg("ecm"),
+      py::arg("name"),
+      "Get the ID of a light entity which is an immediate child of"
+      "this world.")
```
Comment (ahcorde) [inline python/src/gz/sim/World.cc:66]:
```suggestion
      "Get the ID of a light entity which is an immediate child of "
      "this world.")
```

File: python/src/gz/sim/World.cc [hunk 3]
```diff
@@ -22,23 +22,79 @@
...
+  .def("actor_by_name", &gz::sim::World::ActorByName,
+      py::arg("ecm"),
+      py::arg("name"),
+      "Get the ID of a actor entity which is an immediate child of"
+      "this world.")
+  .def("model_by_name", &gz::sim::World::ModelByName,
+      py::arg("ecm"),
+      py::arg("name"),
+      "Get the ID of a model entity which is an immediate child of "
+      "this world.")
```
Comment (ahcorde) [inline python/src/gz/sim/World.cc:76]:
```suggestion
      "Get the ID of a actor entity which is an immediate child of "
      "this world.")
  .def("model_by_name", &gz::sim::World::ModelByName,
      py::arg("ecm"),
      py::arg("name"),
      "Get the ID of a model entity which is an immediate child of "
      "this world.")
```

File: python/src/gz/sim/World.cc [hunk 4]
```diff
@@ -22,23 +22,79 @@
...
+      "Get the number of lights which are immediate children of this"
+      "world.")
+  .def("actor_count", &gz::sim::World::ActorCount,
+      py::arg("ecm"),
+      "Get the number of actors which are immediate children of this"
+      "world.")
+  .def("model_count", &gz::sim::World::ModelCount,
+      py::arg("ecm"),
+      "Get the number of models which are immediate children of this"
+      "world.");
```
Comment (ahcorde) [inline python/src/gz/sim/World.cc:97]:
```suggestion
  .def("light_count", &gz::sim::World::LightCount,
      py::arg("ecm"),
      "Get the number of lights which are immediate children of this "
      "world.")
  .def("actor_count", &gz::sim::World::ActorCount,
      py::arg("ecm"),
      "Get the number of actors which are immediate children of this "
      "world.")
  .def("model_count", &gz::sim::World::ModelCount,
      py::arg("ecm"),
      "Get the number of models which are immediate children of this "
      "world.");
```

File: python/test/world_TEST.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        # Missing a way to set up the atmosphere for the world
+
+    def test_world_magnetic_field(self):
+        self.assertEqual(None, self.world.magnetic_field(self.ecm))
+        # Missing a way to set up the magnetic field for the world
+
+    def test_world_spherical_coordinates(self):
+        self.assertEqual(None, self.world.spherical_coordinates(self.ecm))
+        self.world.set_spherical_coordinates(self.ecm, SphericalCoordinates())
+        # This assertion should be `self.assertEqual(SphericalCoordinates(), self.world.spherical_coordinates(self.ecm))`
```
Comment (ahcorde) [inline python/test/world_TEST.py:54]: `> 100`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: python/src/gz/sim/World.cc (hunk 4 above)
Comment (Voldivh) [reply python/src/gz/sim/World.cc]: Done.

File: python/src/gz/sim/World.cc (hunk 3 above)
Comment (Voldivh) [reply python/src/gz/sim/World.cc]: Done.

File: python/src/gz/sim/World.cc (hunk 2 above)
Comment (Voldivh) [reply python/src/gz/sim/World.cc]: Done.

File: python/src/gz/sim/World.cc (hunk 1 above)
Comment (Voldivh) [reply python/src/gz/sim/World.cc]: Done.

File: python/test/world_TEST.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+
+from gz.common import set_verbosity
+from gz.sim8 import TestFixture, World, world_entity
```
[the remark this answers is not in the document]
Comment (Voldivh) [reply python/test/world_TEST.py]: Done.

File: python/src/gz/sim/World.cc [hunk 7]
```diff
@@ -22,23 +22,79 @@
 
 #include "World.hh"
 
+namespace py = pybind11;
```
[the remark this answers is not in the document]
Comment (Voldivh) [reply python/src/gz/sim/World.cc:23]: Done.

File: CMakeLists.txt [hunk 8]
```diff
@@ -169,6 +169,11 @@ set(GZ_RENDERING_VER ${gz-rendering8_VERSION_MAJOR})
 gz_find_package(gz-math7 REQUIRED COMPONENTS eigen3)
 set(GZ_MATH_VER ${gz-math7_VERSION_MAJOR})
 
+#--------------------------------------
```
[the remark this answers is not in the document]
Comment (Voldivh) [reply CMakeLists.txt]: Done.

File: .github/ci/packages.apt [hunk 9]
```diff
@@ -32,6 +32,7 @@ python3-distutils
 python3-gz-math7
 python3-pybind11
 python3-pytest
+python3-sdformat13
```
Comment (azeey) [inline .github/ci/packages.apt:35]: Update to sdformat14

**Review by azeey — APPROVED** [review]
LGTM! The only remaining changes are to update sdformat13 to sdformat14

File: .github/ci/packages.apt (hunk 9 above)
Comment (Voldivh) [reply .github/ci/packages.apt]: Nice catch. Done.

**Pull request merged**